### PR TITLE
add SkillValue and SkillAllUsersStats

### PR DESCRIPTION
**Pull request by sangaran10**
## Description of changes
The entities SkillValue and SkillAllUsersStats have been added, and will be resolved as new attributes for the skill in the gateway. These entities are available via internal queries. SkillValue combines the six SkillLevelEntity values (Remember, Understand, Apply, Analyse, Evaluate and Create) for each skill into one value. SkillAllUsersStats sums up the SkillValues for all users in the course who have worked on the skill, and it saves the participantCount, which shows how many users have worked on the skill so far. SkillAllUsersStats will be used in the gateway to determine the average skill value score.

  
## How has this been tested?
Please describe the test strategy you followed.

- [ ] automated unit test
- [ ] automated integration test
- [ ] automated acceptance test
- [ ] manual, exploratory test

In case of manual test, please document the test well including a set of user instructions and prerequisites. Each including an action, it's result, and where appropriate a screenshot.
In case of automated tests, the test coverage should be reasonably high, especially on Service classes and other classes with complex logic.
    
## Checklist before requesting a review
- [ ] My code follows the [coding guidelines](https://github.com/MEITREX/wiki/blob/main/dev-manuals/backend/coding-guidelines.md) of this project
- [ ] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] My code fulfills all acceptance criteria
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] The test coverage (line and branch) is reasonably high, especially on Service classes and other classes with complex logic
- [ ] I have made corresponding changes to the documentation
- [ ] I have added explanation of architectural decision and rationales to [wiki/adr](https://github.com/MEITREX/wiki/tree/main/adr)
- [ ] I have updated the changes in the ticket description

## Checklist for reviewer
- [ ] The code works and does not throw errors
- [ ] The code is easy to understand and there are no confusing parts
- [ ] The code follows the [coding guidelines](https://github.com/MEITREX/wiki/blob/main/dev-manuals/backend/coding-guidelines.md) of this project
- [ ] The code change accomplishes what it is supposed to do
- [ ] I cannot think of any use case in which the code does not behave as intended
- [ ] The added and existing tests reasonably cover the code change
- [ ] I cannot think of any test cases, input or edge cases that should be tested in addition
- [ ] Description of the change is included in the documentation

### CODE REVIEW

**Review by MP-DG — CHANGES_REQUESTED** [review]
Please write testcases for your changes, currently 0% coverage

File: src/main/java/de/unistuttgart/iste/meitrex/skilllevel_service/service/SkillLevelService.java [hunk 1]
```diff
@@ -77,7 +81,94 @@ public List<SkillLevels> getSkillLevelsForCourse(UUID courseId, UUID userId) {
...
+        for(Float skillValue : skillValues) {
+            skillValueSum += skillValue;
+        }
+        SkillAllUsersStatsEntity skillAllUsersStats = new SkillAllUsersStatsEntity();
+        skillAllUsersStats.setSkillId(skillId);
+        skillAllUsersStats.setParticipantCount(skillValues.size());
+        if (skillValues.isEmpty()) {
+            skillAllUsersStats.setSkillValueSum(0f);
+        } 
+        else {
```
Comment (MP-DG) [inline src/main/java/de/unistuttgart/iste/meitrex/skilllevel_service/service/SkillLevelService.java]: Does not look formatted well

File: src/main/java/de/unistuttgart/iste/meitrex/skilllevel_service/service/SkillLevelService.java [hunk 2]
```diff
@@ -77,7 +81,94 @@ public List<SkillLevels> getSkillLevelsForCourse(UUID courseId, UUID userId) {
...
+            skillLevel.getApply().getValue(),
+            skillLevel.getAnalyze().getValue(),
+            skillLevel.getEvaluate().getValue(),
+            skillLevel.getCreate().getValue()
+        );
+        List<Float> nonZeroValues = values.stream().filter(v -> v > 0f).toList();
+        if (nonZeroValues.isEmpty()) {
+            return 0f;
+        }
+        else{
```
Comment (MP-DG) [inline src/main/java/de/unistuttgart/iste/meitrex/skilllevel_service/service/SkillLevelService.java]: Same here

File: src/main/java/de/unistuttgart/iste/meitrex/skilllevel_service/service/SkillLevelService.java [hunk 3]
```diff
@@ -77,7 +81,94 @@ public List<SkillLevels> getSkillLevelsForCourse(UUID courseId, UUID userId) {
...
+     * return the skill values for the given user and skills
+     *
+     * @param skillIds the ids of the skills
+     * @param userId   the id of the user
+     * @return the skill values for the given user and skills
+     */
+    public List<SkillValueEntity> getSkillValuesForSkillIds(List <UUID> skillIds, UUID userId) {
+        List<AllSkillLevelsEntity> skillLevels = getSkillLevelEntitiesForSkillIds(skillIds, userId);
+        List<SkillValueEntity> skillValues = new ArrayList<>();
+        for(AllSkillLevelsEntity skillLevel : skillLevels) {
```
Comment (MP-DG) [inline src/main/java/de/unistuttgart/iste/meitrex/skilllevel_service/service/SkillLevelService.java]: missing space after for

File: src/main/java/de/unistuttgart/iste/meitrex/skilllevel_service/service/SkillLevelService.java [hunk 4]
```diff
@@ -77,7 +81,94 @@ public List<SkillLevels> getSkillLevelsForCourse(UUID courseId, UUID userId) {
...
+     * like skillValue sum and participantCount of the given skills
+     */
+    private SkillAllUsersStatsEntity getSkillAllUsersStatsForSkillId(UUID skillId) {
+        List<AllSkillLevelsEntity> skillLevels = skillLevelsRepository.findByIdSkillId(skillId);
+        List<Float> skillValues = new ArrayList<>();
+        for(AllSkillLevelsEntity skillLevel : skillLevels) {
+            skillValues.add(getSkillValueForSkillLevel(skillLevel));
+        }
+        float skillValueSum = 0f;
+        for(Float skillValue : skillValues) {
```
Comment (MP-DG) [inline src/main/java/de/unistuttgart/iste/meitrex/skilllevel_service/service/SkillLevelService.java]: missing space after for

File: src/main/java/de/unistuttgart/iste/meitrex/skilllevel_service/service/SkillLevelService.java [hunk 5]
```diff
@@ -77,7 +81,94 @@ public List<SkillLevels> getSkillLevelsForCourse(UUID courseId, UUID userId) {
...
+    /**
+     * Returns the stats for the skills for all users from the course of the skills.
+     *
+     * @param skillIds List of skillIds
+     * @return List of SkillAllUsersStatsEntity that represents stats
+     * like skillValueSum and participantCount of the given skills
+     */
+    public List<SkillAllUsersStatsEntity> getSkillsAllUsersStatsForSkillIds(List<UUID> skillIds) {
+        List<SkillAllUsersStatsEntity> skillAllUsersStats = new ArrayList<>();
+        for(UUID skillId : skillIds) {
```
Comment (MP-DG) [inline src/main/java/de/unistuttgart/iste/meitrex/skilllevel_service/service/SkillLevelService.java]: missing space after for

**Review by MP-DG — CHANGES_REQUESTED** [review]
Code is not formatted according to project style guide in 
src/main/java/de/unistuttgart/iste/meitrex/skilllevel_service/service/SkillLevelService.java

Besides that, it looks good

**Review by MP-DG — APPROVED** [review]
good now
